iocage/lib/Config/Jail/Defaults.py

```diff
@@ -26,11 +26,38 @@
 import iocage.lib.Config.Jail.BaseConfig
 
 
-class JailConfigDefaults(iocage.lib.Config.Jail.BaseConfig.BaseConfig):
+class DefaultsUserData(dict):
```

Comment:
can you briefly explain what happened.

Reply:
JailConfigDefaults should return all default properties as `data`, but only the user overrides as `user_data`, so that only manual overrides are saved to `defaults.json`. It was not sufficient to provide a reduced copy of self.data as user_data because some properties were references. DefaultsUserData goes one level deeper and intercepts setting individual attributes, so that we no longer need to pass the reduced copy of data.